### PR TITLE
[MMI] Fixed bug that was causing to not show the correct account name

### PR DESCRIPTION
**Pull request by albertolive**
## Explanation

Fixed bug causing incorrect display of account name; Now shows the correct name instead of the default "Account x".

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by shane-t — APPROVED** [review]

File: app/scripts/controllers/mmi-controller.js [hunk 1]
```diff
@@ -292,9 +292,9 @@ export default class MMIController extends EventEmitter {
       })),
     );
 
-    newAccounts.forEach(
-      async () => await this.keyringController.addNewAccount(keyring),
-    );
+    for (let i = 0; i < newAccounts.length; i++) {
```
Comment (albertolive) [inline app/scripts/controllers/mmi-controller.js:295]: The forEach method does not wait for the promises returned by the addNewAccount method to resolve before moving on to the next iteration. This means that all calls to addNewAccount will be made concurrently, which could cause issues. 

TLDR; In this case, the code was continuing, so `syncAddresses` was called before `setAccountLabel`.

File: app/scripts/controllers/mmi-controller.js [hunk 2]
```diff
@@ -303,12 +303,33 @@ export default class MMIController extends EventEmitter {
       ...new Set(oldAccounts.concat(allAccounts.map((a) => a.toLowerCase()))),
     ];
 
+    // Create a Set of lowercased addresses from oldAccounts for efficient existence checks
```
Comment (albertolive) [inline app/scripts/controllers/mmi-controller.js:306]: Improved performance below

**Review by DDDDDanica — APPROVED** [review]
